### PR TITLE
[docs] document CUDA version support

### LINKED ISSUE #3416: Traning model on multi GPUs?
I prefer to use multi GPUs to speed up traning model, but there is only few information about this in the document. Does lightGBM works well on multi GPUs ?

### PR DESCRIPTION
**Pull request by StrikerRUS**
In follow up of #3160.

Closed #3416.

cc @ChipKerchner

### CODE REVIEW

File: docs/Installation-Guide.rst
Comment (jameslamb) [inline docs/Installation-Guide.rst:568]:
```suggestion
The `original GPU build <#build-gpu-version>`__ of LightGBM is based on OpenCL.

The CUDA-based build is a separate implementation and requires an NVIDIA graphics card with compute capability 6.0 and higher. It should be considered experimental, and we suggest using it only when it is impossible to use OpenCL version (for example, on IBM POWER microprocessors).

**Note**: only Linux is supported, other operating systems are not supported yet.
```

When I read this, I was really confused whether "this version" refers to OpenCL or CUDA. Could we clarify it?

File: docs/Installation-Guide.rst
Comment (jameslamb) [inline docs/Installation-Guide.rst:577]: is there some documentation from NVIDIA that we can link to here? I think "install CUDA libraries" is too vague.

Maybe something like [this](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html)?

File: docs/Installation-Guide.rst
Comment (jameslamb) [inline docs/Installation-Guide.rst:588]: I have a slight preference for putting each command on its own line, instead of grouping them with `;`. That makes it a little easier to understand at first glance, in my opinion.

```suggestion
  git clone --recursive https://github.com/microsoft/LightGBM
  cd LightGBM
  mkdir build
  cd build
  cmake -DUSE_CUDA=1 ..
  make -j4
```

What do you think? If you disagree, it's ok to leave it as-is. It's not worth holding up the PR.

File: docs/Parameters.rst
Comment (jameslamb) [inline docs/Parameters.rst:1129]:
```suggestion
   -  **Note**: can be used only in OpenCL implementation, in CUDA implementation only double precision is currently supported
```

File: include/LightGBM/config.h
Comment (jameslamb) [inline include/LightGBM/config.h:971]:
```suggestion
  // desc = **Note**: can be used only in OpenCL implementation, in CUDA implementation only double precision is currently supported
```

File: python-package/README.rst
Comment (jameslamb) [inline python-package/README.rst:120]:
```suggestion
All requirements from `Build from Sources section <#build-from-sources>`__ apply for this installation option as well, and `CMake`_ (version 3.16 or higher) is strongly required.
```

File: python-package/README.rst
Comment (jameslamb) [inline python-package/README.rst:180]:
```suggestion
Run ``python setup.py install --cuda`` to enable CUDA support. All requirements from `Build CUDA Version section <#build-cuda-version>`__ apply for this installation option as well.
```

**Review by jameslamb — CHANGES_REQUESTED** [review]
thanks for doing this! I left a few suggestions on clarifying the language, but otherwise I agree with these changes.

File: docs/Installation-Guide.rst
Comment (StrikerRUS) [reply docs/Installation-Guide.rst:588]: I totally agree! But could it be done in a separate PR to change all occurrences of such commands across the doc?

File: python-package/README.rst
Comment (StrikerRUS) [reply python-package/README.rst:120]: This is standard phrase across this doc. Again, if you do not mind I'll change it in a separate PR to keep the doc consistent.

File: python-package/README.rst
Comment (StrikerRUS) [reply python-package/README.rst:180]: The same about a separate PR to change the wording of all paragraphs at once.

File: docs/Installation-Guide.rst
Comment (jameslamb) [reply docs/Installation-Guide.rst:588]: oh sure, no problem. I didn't know there were other places that do this. It can be another PR, and not urgent at all

File: python-package/README.rst
Comment (jameslamb) [reply python-package/README.rst:120]: ah ok no problem. I was just taking the diff as-is...didn't realize some other these things were patterns reused elsewhere in the doc. Changing them in a different PR is ok and not urgent.

File: docs/Installation-Guide.rst
Comment (StrikerRUS) [reply docs/Installation-Guide.rst]: Fixed in 4a175a9625fe2542abb600bcd6647927157b4dbe.

File: docs/Installation-Guide.rst
Comment (StrikerRUS) [reply docs/Installation-Guide.rst]: Fixed in 4a175a9625fe2542abb600bcd6647927157b4dbe.

File: docs/Parameters.rst
Comment (StrikerRUS) [reply docs/Parameters.rst]: Fixed in 4a175a9625fe2542abb600bcd6647927157b4dbe.

File: include/LightGBM/config.h
Comment (StrikerRUS) [reply include/LightGBM/config.h]: Fixed in 4a175a9625fe2542abb600bcd6647927157b4dbe.

File: docs/Installation-Guide.rst
Comment (StrikerRUS) [reply docs/Installation-Guide.rst:588]: Thanks! I'll prepare a new PR right after merging this one.

**Review by jameslamb — APPROVED** [review]
looks good, thank you for this!

**Comment by StrikerRUS** [comment]
Depends on fixing CUDA error reported in #3450 and https://github.com/microsoft/LightGBM/pull/3424#issuecomment-702189313.

**Comment by bordaw** [comment]
Do you know when the official LightGBM documentation will be updated to describe the CUDA support?

**Comment by StrikerRUS** [comment]
@bordaw Right after fixing critical bugs of the initial CUDA implementation: https://github.com/microsoft/LightGBM/pull/3428#issuecomment-721307376. Unfortunately, it seems that it won't be very quickly: https://github.com/microsoft/LightGBM/pull/3160#issuecomment-745338726.

**Comment by bordaw** [comment]
Thanks for the prompt response! Please let me know if I can help in the fix as I am one of the original authors of the CUDA implementation.

**Comment by StrikerRUS** [comment]
@bordaw Wow, this will be just awesome! Thank you very much for wishing to help!

I think you can try to reproduce the issue locally first: https://github.com/microsoft/LightGBM/pull/3160#issuecomment-724301806. Please let me know if you need any additional details.

**Comment by bordaw** [comment]
Thanks! Will download the latest code and take a look..

**Comment by StrikerRUS** [comment]
@jameslamb @guolinke Please (re-)review when have time.

**Review by jameslamb — APPROVED** [review]
I have no changes to suggest, this looks good to me. Thanks for adding the explanation on how the CUDA build is different from OpenCL and why you might use it. Really helpful.

**Pull request merged**
